jgrapht-core/src/main/java/org/jgrapht/alg/cycle/AhujaOrlinSharmaCyclicExchangeLocalAugmentation.java

```diff
@@ -0,0 +1,156 @@
+
+        // initialize PathsLengthK for k = 1
+        for(E e : graph.edgeSet()) {
+            if(graph.getEdgeWeight(e) < 0) {
+                // initialize all paths of cost < 0
+                LinkedList<V> pathVertices = new LinkedList<>();
+                Map<V, Integer> pathLabels = new LinkedHashMap<>();
+                V sourceVertex = graph.getEdgeSource(e);
+                V targetVertex = graph.getEdgeTarget(e);
+                pathVertices.add(sourceVertex);
+                pathVertices.add(targetVertex);
+                pathLabels.put(sourceVertex, labels.get(sourceVertex));
+                pathLabels.put(targetVertex, labels.get(targetVertex));
+                LabeledPath<V> path = new LabeledPath<>(pathVertices, graph.getEdgeWeight(e), pathLabels);
+
+                // add path to set of paths of length 1
+                PathsLengthK.add(path);
+            }
+        }
+
+        while(k < lengthBound && C.getCost() >= 0) {
+            while(!PathsLengthK.isEmpty()) {
+                for(Iterator<LabeledPath<V>> it = PathsLengthK.iterator(); it.hasNext();) { // go through all valid paths of length k
+                    LabeledPath<V> path = it.next();
+                    it.remove();
+
+                    V head = path.getHead();
+                    V tail = path.getTail();
+
+                    if(graph.containsEdge(tail, head) && path.getCost() + graph.getEdgeWeight(graph.getEdge(tail, head)) < C.getCost()) { // the path builds a valid cycle
+                        C = path.clone();
+                        C.addVertex(head, graph.getEdgeWeight(graph.getEdge(tail, head)), labels.get(head));
+                        if(C.getCost() < 0) { // only return the cycle if it is negative (in the first iteration it can be non-negative)
+                            return C;
+                        }
+                    }
+
+                    for(E e : graph.outgoingEdgesOf(tail)) {
+                        V currentVertex = graph.getEdgeTarget(e);
+                        if(!path.getLabels().contains(labels.get(currentVertex)) && path.getCost() + graph.getEdgeWeight(e) < 0) { // extend the path if the extension is still negative a correctly labeled
+                            LabeledPath<V> newPath = path.clone();
+                            newPath.addVertex(currentVertex, graph.getEdgeWeight(e), labels.get(currentVertex));
+                            PathsLengthKplus1.add(newPath);
+
+                            testDomination(path,  PathsLengthKplus1); // check if paths are dominated, i.e. if the path is definitely worse than other paths and does not have to be considered in the future
+                        }
+                    }
+                }
+            }
+            // update k and the corresponding sets
+            k += 1;
+            PathsLengthK = PathsLengthKplus1;
+            PathsLengthKplus1 = new LinkedHashSet<>();
+        }
+        return null;
```

Comment:
I think you should state in the documentation that `null` is returned if no such cycle is found.

Reply:
thanks, I removed them by an empty path